pos_localstorage/static/src/js/models.js

```diff
@@ -0,0 +1,224 @@
+            // List of product attributes that the POS uses.
+            var productAttributeList = ['name', 'list_price','price','pos_categ_id', 'taxes_id', 'ean13', 'default_code', 'variants',
+                     'to_weight', 'uom_id', 'uos_id', 'uos_coeff', 'mes_type', 'description_sale', 'description'];
+
+            var lastUpdate = pos.db.load('last_product_update');
+            if (!lastUpdate)
+                lastUpdate = '1970-01-01T00:00:00Z';
+
+            // Fetch all products that we sell and that have not been changed since
+            //  we updated for the last time.
+            return pos.fetch(
+                'product.product',
+                ['id'],
+                [['sale_ok', '=', true], ['available_in_pos', '=', true], ['write_date', '<', lastUpdate]]
+            ).then(function(products){
+                var products_to_add = [];
+                var products_to_fetch = [];
+                for (var i in products) {
+                    // Try to get product information from localStorage.
+                    // If this fails, we load the information from the server.
+                    var product = pos.db.load('product_' + products[i].id);
+                    if (product) {
+                        products_to_add.push(product);
+                    } else {
+                        products_to_fetch.push(products[i].id);
+                    }
+                }
+                // Add products that we have in localStorage
+                pos.db.add_products(products_to_add);
+
+                // Fetch all products that we don't have the data
+                //  because they were not in localStorage or they are new
+                return pos.fetch('product.product',
+                                 productAttributeList,
+                                 ['|', ['id', 'in', products_to_fetch],
+                                       '&', '&', ['sale_ok', '=', true], ['available_in_pos', '=', true], ['write_date', '>=', lastUpdate]],
+                                 {pricelist: pos.get('shop').pricelist_id[0]} // context for price
+                );
+            }).then(function(products){
+                pos.db.add_products(products);
+
+                lastUpdate = new Date(Date.now() - _searchGracePeriod).toISOString();
+                pos.db.save('last_product_update', lastUpdate);
+            });
+        },
+    };
+
+
+    var _old_initialize = module.PosModel.prototype.initialize;
+    module.PosModel.prototype.initialize = function(session, attributes) {
+        var ret = _old_initialize.call(this, session, attributes);
+        // HACK to make localStorage abstraction dependent of the database
+        this.db.name = session.db;
+        // clear database again after setting the name;
+        this.db.clear('products','categories');
+        return ret;
+    };
+
+    // Overwrite method to additionally save product data in localStorage
```

Comment:
in fact it's 'overload', not 'overwrite' ? because you call super in your function...